PSPDFKitPlugin/PSPDFKitPlugin.m

```diff
@@ -1438,4 +1373,229 @@ - (CGRect)flexibleToolbarContainerContentRect:(PSPDFFlexibleToolbarContainer *)c
+        success = [document addAnnotations:@[annotation] options:nil];
+    }
+
+    if (success) {
+        [self.commandDelegate sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_OK]
+                                    callbackId:command.callbackId];
+    } else {
+        NSLog(@"Failed to add annotation.");
+        [self.commandDelegate sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR]
+                                    callbackId:command.callbackId];
+    }
+}
+
+- (void)removeAnnotation:(CDVInvokedUrlCommand *)command {
+    id jsonAnnotation = [command argumentAtIndex:0];
+    NSString *annotationUUID = jsonAnnotation[@"uuid"];
+    if (annotationUUID.length == 0) {
+        NSLog(@"Invalid annotation UUID.");
+        [self.commandDelegate sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR]
+                                    callbackId:command.callbackId];
+    }
+    
+    PSPDFDocument *document = self.pdfController.document;
+    VALIDATE_DOCUMENT(document)
+    BOOL success = NO;
+
+    NSArray<PSPDFAnnotation *> *allAnnotations = [[document allAnnotationsOfType:PSPDFAnnotationTypeAll].allValues valueForKeyPath:@"@unionOfArrays.self"];
+    for (PSPDFAnnotation *annotation in allAnnotations) {
+        // Remove the annotation if the uuids match.
+        if ([annotation.uuid isEqualToString:annotationUUID]) {
+            success = [document removeAnnotations:@[annotation] options:nil];
+            break;
+        }
+    }
+
+    if (success) {
+        [self.commandDelegate sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_OK]
+                                    callbackId:command.callbackId];
+    } else {
+        NSLog(@"Failed to remove annotation.");
+        [self.commandDelegate sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR]
+                                    callbackId:command.callbackId];
+    }
+}
+
+- (void)getAllUnsavedAnnotations:(CDVInvokedUrlCommand *)command {
+    PSPDFDocument *document = self.pdfController.document;
+    VALIDATE_DOCUMENT(document)
+
+    PSPDFDocumentProvider *documentProvider = document.documentProviders.firstObject;
+    NSData *data = [document generateInstantJSONFromDocumentProvider:documentProvider error:NULL];
+    NSDictionary *annotationsJSON = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:NULL];
+    CDVPluginResult *pluginResult;
+    if (annotationsJSON) {
+        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:annotationsJSON];
+    }  else {
+        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:@{@"localizedDescription": @"Failed to get annotations"}];
+    }
+
+    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+}
+
+- (void)addAnnotations:(CDVInvokedUrlCommand *)command {
```

Comment:
`addAnnotation` uses the [Instant JSON Annotation API](https://pspdfkit.com/guides/ios/current/importing-exporting/instant-json/#instant-annotation-json-api), while `addAnnotations` uses the [Instant JSON Document API](https://pspdfkit.com/guides/ios/current/importing-exporting/instant-json/#instant-document-json-api). We have the same API in our React Native wrapper. See and https://github.com/PSPDFKit/react-native/blob/master/index.js#L180 and https://github.com/PSPDFKit/react-native/blob/master/index.js#L249

Reply:
The Document API also contains bookmarks - maybe really do applyInstantJSON?

Reply:
That's a very good point! The reasons why I opted for `addAnnotations` despite this, are:

1. Consistency with React Native naming.
2. `addAnnotations` sounds more intuitive to the end user.

What do you think?

Reply:
I really vote for `applyInstantJSON` here, since it's annotations, forms and bookmarks (and potentially more in the future).

We should fix the React name as well.

https://pspdfkit.com/guides/ios/current/importing-exporting/instant-json/

Reply:
Thank you! I renamed `addAnnotations` to `applyInstantJSON` in https://github.com/PSPDFKit/Cordova-iOS/pull/75/commits/69e4767273ef484af7d1bbc5925fcbe9eec92d56